include/azure_kinect_ros_driver/k4a_exposure_calibration.h

```diff
@@ -0,0 +1,158 @@
+     * @param[out] res_msg human-readable error message included in response
+     */
+    bool k4aBlueBoundsCheck(int target_blue_value, int& error_code, std::string& res_msg);
+    
+    /**
+     * @brief did the node receive an image at all?
+     * @param[in] mat current OpenCV mat stored in latest_k4a_image for k4aAutoTuneExposure loop
+     * @param[out] error_code error code included in response
+     * @param[out] res_msg human-readable error message included in response
+     */
+    bool k4aImagePopulatedCheck(cv::Mat& mat, int& error_code, std::string& res_msg);
+
+private:
+    /**
+     * @brief callback for /rgb/raw/image subscription
+     * @details This callback handles updating the current image for the node.
+     *          It converts the ROS image message to an OpenCV mat for auto tuning calculations.
+     * @param[out] msg message received from image transport subscriber subscribed to /rgb/raw/image
+     */
+    void rgbRawImageCallback(const sensor_msgs::ImageConstPtr& msg);
+
+    /**
+     * @brief callback for k4aUpdateExposure
+     * @details This callback handles requests to update the camera exposure from the
+     *          k4a_update_exposure service.
+     * @param[in] req request received from calling k4a_update_exposure
+     * @param[out] res response sent from k4a_update_exposure
+     */
+    bool k4aUpdateExposureCallback(azure_kinect_ros_driver::k4a_update_exposure::Request &req,
+                                   azure_kinect_ros_driver::k4a_update_exposure::Response &res);
+    
+    /**
+     * @brief callback for k4aAutoTuneExposure
+     * @details This callback handles requests to update the camera exposure from the
+     *          k4a_auto_tune_exposure service.
+     * @param[in] req request received from calling k4a_auto_tune_exposure
+     * @param[out] res response sent from k4a_auto_tune_exposure
+     */
+    bool k4aAutoTuneExposureCallback(azure_kinect_ros_driver::k4a_auto_tune_exposure::Request &req,
+                                     azure_kinect_ros_driver::k4a_auto_tune_exposure::Response &res);
+
+    // private members
+    ros::NodeHandle nh_;
+    image_transport::Subscriber subRGBRaw;
+    ros::ServiceServer update_exposure_service;
+    ros::ServiceServer auto_tune_exposure_service;
+
+    // allocate memory space to store latest image
+    cv::Mat latest_k4a_image; /** @brief latest image*/
+    cv::Mat* const latest_k4a_image_ptr = &latest_k4a_image; /** @brief pointer to latest image*/
+    cv_bridge::CvImageConstPtr k4aCvImagePtr; /** @brief pointer to convert ROS image to OpenCV mat*/
+    std::mutex latest_k4a_image_mutex; /** @brief mutex to protect latest image*/
+    azure_kinect_ros_driver::k4aCameraExposureServiceErrorCode k4a_error_code; /** @brief error codes*/
+
+    // config file info
+    // TODO: PULL THESE VALUES FROM CONFIG FILE "azure_kinect_ros_driver/AzureKinectParamsConfig.h"
+    const int MIN_EXPOSURE = 488;
```

Comment:
use uint32_t or uint64_t instead of int. Make it unsigned and either 32 bit or 64 bit. Up to you.